File: python/fibonacci.py
	
class fibonacci:
	def __init__(self):
		self.s1 = 1
		self.s2 = 1
		
	def next(self):
		(self.s1, self.s2, current) = (self.s1 + self.s2, self.s1, self.s2)
		return current
		
	def __iter__(self):
		return self
		

	

File: js/fibonacci.js
var Fibonacci = function (first, second) {
    return {
        first: first,
        second: second,
        nextToLast: first,
        last: second,

        sequence: [first, second],

        next: function () {
            var x = this.nextToLast + this.last;
            this.nextToLast = this.last;
            this.last = x;
            this.sequence.push(this.last);
            return x;
        }
    }
};

module.exports.Fibonacci = Fibonacci;

